File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        dp = [0 for i in range(n+1)]
        for i in range(n, -1, -1):
            if i in (n, n-1):
                dp[i] = 1
            else:
                dp[i] = dp[i+1]+dp[i+2]
        print(dp)
        return dp[0]

        

File: 0070-climbing-stairs/0070-climbing-stairs.java
class Solution {
    public int climbStairs(int n) {
        if(n < 2){
            return n;
        }
        int i = 1, n1 = 0, n2 = 1;
        int n3 = n1 + n2;
        while(i < n){
            n1 = n2;
            n2 = n3;
            n3 = n1+n2;
            i++;
        }
        return n3;

    }
}